python/argo_client/connection.py

```diff
@@ -158,32 +158,22 @@ def __init__(self, command: str, *,
         self.persist = persist
         super().__init__(command, environment=environment)
 
-
-    def buffer_replies(self) -> None:
-        """Read any replies that the server has sent, and add their byte
-           representation to the internal buffer, freeing up space in
-           the pipe or socket.
-        """
-        try:
-            arrived = self.socket.recv(4096)
-            while arrived != b'':
-                self.buf.extend(arrived)
-                arrived = self.socket.recv(4096)
-            return None
-        except BlockingIOError:
-            return None
```

Comment:
Since we deleted a method from this class, this PR technically changes the `argo-client` API. It would be worth documenting this change in the changelog (as well as the removal of `RemoteSocketProcess.buffer_replies` and `ServerConnection._process_replies`).

(For what it's worth, I checked and neither `cryptol` nor `saw-client` appear to use any of these methods directly, so I doubt there will be any breakage in practice.)